<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1300</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">*{font: normal 20px “微软雅黑” ;}

QWidget#centralwidget {
	background-color: #eaeaea;
}

.QLabel {
	padding: 10px;
	border-radius: 5px;
}

.QPushButton {
	border: 1px solid;
	background-color:qlineargradient(spread:pad,x1:0,y1:0,x2:0,y2:1,stop:0 #e9e9e9,stop:1 #f2f2f3);
	padding: 10px;
	border-radius: 5px;
}

.QPushButton:hover {
    background-color: #ecf5ff;
    color: #409eff;
}

.QPushButton:pressed, QPushButton:checked {
    border: 1px solid #3a8ee6;
    color: #409eff;
}

.QTableWidget{
	color: #000000;
	alternate-background-color: #e0e0e0;
}

.QTableWidget::item:selected {
	color: #000000; 
	background-color: #8fcde6;

}

.QComboBox {
	background-color:qlineargradient(spread:pad,x1:0,y1:0,x2:0,y2:1,stop:0 #e9e9e9,stop:1 #f2f2f3);
  	padding: 10;
 	border:1px solid rgba(228,228,228,1);
	border-radius:5px;
}

.QLineEdit {
	border-radius:5px;
	padding: 10;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,4">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>串口模式</string>
        </property>
        <layout class="QFormLayout" name="formLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="font">
            <font>
             <family>“微软雅黑”</family>
             <pointsize>-1</pointsize>
             <weight>50</weight>
             <italic>false</italic>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>串口号：</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="comboBox_com">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="font">
            <font>
             <family>“微软雅黑”</family>
             <pointsize>-1</pointsize>
             <weight>50</weight>
             <italic>false</italic>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>波特率：</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QComboBox" name="comboBox_rate">
           <item>
            <property name="text">
             <string>9600</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>38400</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>115200</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="2" column="0">
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>94</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="pushButton_SerialOpen">
           <property name="font">
            <font>
             <family>“微软雅黑”</family>
             <pointsize>-1</pointsize>
             <weight>50</weight>
             <italic>false</italic>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>打开串口</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>TCP模式</string>
        </property>
        <layout class="QFormLayout" name="formLayout_2">
         <property name="sizeConstraint">
          <enum>QLayout::SetDefaultConstraint</enum>
         </property>
         <property name="fieldGrowthPolicy">
          <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
         </property>
         <property name="rowWrapPolicy">
          <enum>QFormLayout::DontWrapRows</enum>
         </property>
         <property name="labelAlignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
         <property name="formAlignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label_sIP">
           <property name="font">
            <font>
             <family>“微软雅黑”</family>
             <pointsize>-1</pointsize>
             <weight>50</weight>
             <italic>false</italic>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>本地IP：</string>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QComboBox" name="comboBox_IP"/>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_sPort">
           <property name="font">
            <font>
             <family>“微软雅黑”</family>
             <pointsize>-1</pointsize>
             <weight>50</weight>
             <italic>false</italic>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>本地端口：</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="lineEdit_sPort"/>
         </item>
         <item row="2" column="0">
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>79</width>
             <height>25</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="pushButton_Listen">
           <property name="font">
            <font>
             <family>“微软雅黑”</family>
             <pointsize>-1</pointsize>
             <weight>50</weight>
             <italic>false</italic>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>监听</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="1" column="1">
         <widget class="QPushButton" name="pushButton_DeleteSample">
          <property name="font">
           <font>
            <family>“微软雅黑”</family>
            <pointsize>-1</pointsize>
            <weight>50</weight>
            <italic>false</italic>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>删除样本号</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QTableWidget" name="tableWidget_sampleNo"/>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="pushButton_AddSample">
          <property name="font">
           <font>
            <family>“微软雅黑”</family>
            <pointsize>-1</pointsize>
            <weight>50</weight>
            <italic>false</italic>
            <bold>false</bold>
           </font>
          </property>
          <property name="text">
           <string>添加样本号</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>测试结果</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="1" column="0">
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>281</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="pushButton_ClearTestResult">
           <property name="font">
            <font>
             <family>“微软雅黑”</family>
             <pointsize>-1</pointsize>
             <weight>50</weight>
             <italic>false</italic>
             <bold>false</bold>
            </font>
           </property>
           <property name="text">
            <string>清空测试结果</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0" colspan="2">
          <widget class="QTableWidget" name="tableWidget_result">
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAsNeeded</enum>
           </property>
           <property name="textElideMode">
            <enum>Qt::ElideRight</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>测试项目</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QTableWidget" name="tableWidget_Test"/>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_AddTest">
             <property name="font">
              <font>
               <family>“微软雅黑”</family>
               <pointsize>-1</pointsize>
               <weight>50</weight>
               <italic>false</italic>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>添加项目</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_DeleteTest">
             <property name="font">
              <font>
               <family>“微软雅黑”</family>
               <pointsize>-1</pointsize>
               <weight>50</weight>
               <italic>false</italic>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>删除项目</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_SendTest">
             <property name="font">
              <font>
               <family>“微软雅黑”</family>
               <pointsize>-1</pointsize>
               <weight>50</weight>
               <italic>false</italic>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>实验下发</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_Edit">
             <property name="font">
              <font>
               <family>“微软雅黑”</family>
               <pointsize>-1</pointsize>
               <weight>50</weight>
               <italic>false</italic>
               <bold>false</bold>
              </font>
             </property>
             <property name="text">
              <string>编辑组合</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_SerialOpen</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_btn_SerialOpen_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>167</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_Listen</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_btn_Listen_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>265</x>
     <y>368</y>
    </hint>
    <hint type="destinationlabel">
     <x>236</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_AddTest</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_btn_AddTest_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>853</x>
     <y>673</y>
    </hint>
    <hint type="destinationlabel">
     <x>747</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_DeleteTest</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_btn_DeleteTest_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>961</x>
     <y>673</y>
    </hint>
    <hint type="destinationlabel">
     <x>747</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_SendTest</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_btn_SendTest_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1069</x>
     <y>673</y>
    </hint>
    <hint type="destinationlabel">
     <x>747</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_Edit</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_btn_Edit_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1177</x>
     <y>673</y>
    </hint>
    <hint type="destinationlabel">
     <x>747</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_ClearTestResult</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_btn_ClearTestResult_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1144</x>
     <y>335</y>
    </hint>
    <hint type="destinationlabel">
     <x>747</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_AddSample</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_btn_AddSample_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>118</x>
     <y>648</y>
    </hint>
    <hint type="destinationlabel">
     <x>117</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_DeleteSample</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_btn_DeleteSample_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>648</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableWidget_Test</sender>
   <signal>cellDoubleClicked(int,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_tableWidget_Test_cellDoubleClicked(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>687</x>
     <y>445</y>
    </hint>
    <hint type="destinationlabel">
     <x>702</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableWidget_Test</sender>
   <signal>cellClicked(int,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_tableWidget_Test_cellClicked(int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>727</x>
     <y>445</y>
    </hint>
    <hint type="destinationlabel">
     <x>747</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableWidget_sampleNo</sender>
   <signal>currentCellChanged(int,int,int,int)</signal>
   <receiver>MainWindow</receiver>
   <slot>slot_tableWidget_sampleNo_currentCellChanged(int,int,int,int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>456</y>
    </hint>
    <hint type="destinationlabel">
     <x>98</x>
     <y>515</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slot_btn_SerialOpen_clicked()</slot>
  <slot>slot_btn_Listen_clicked()</slot>
  <slot>slot_btn_AddTest_clicked()</slot>
  <slot>slot_btn_DeleteTest_clicked()</slot>
  <slot>slot_btn_SendTest_clicked()</slot>
  <slot>slot_btn_Edit_clicked()</slot>
  <slot>slot_btn_ClearTestResult_clicked()</slot>
  <slot>slot_btn_AddSample_clicked()</slot>
  <slot>slot_btn_DeleteSample_clicked()</slot>
  <slot>slot_tableWidget_Test_cellDoubleClicked(int,int)</slot>
  <slot>slot_tableWidget_Test_cellClicked(int,int)</slot>
  <slot>slot_tableWidget_sampleNo_currentCellChanged(int,int,int,int)</slot>
 </slots>
</ui>
